packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/actions.ts

```diff
@@ -26,197 +26,228 @@ import { BankTransferAccountType } from 'data/types'
 
 import { SwapAccountType } from '../swap/types'
 import * as AT from './actionTypes'
-import {
-  SBFixType,
-  SBShowModalOriginType,
-  SimpleBuyActionTypes,
-  StepActionsPayload
-} from './types'
+import { SBFixType, SBShowModalOriginType, SimpleBuyActionTypes, StepActionsPayload } from './types'
+
+const getPayloadObjectForStep = (payload: StepActionsPayload) => {
+  switch (payload.step) {
+    case 'LINKED_PAYMENT_ACCOUNTS':
+    case 'PAYMENT_METHODS':
+      return {
+        cryptoCurrency: payload.cryptoCurrency,
+        fiatCurrency: payload.fiatCurrency,
+        order: payload.order,
+        pair: payload.pair,
+        step: payload.step
+      }
+    case 'VERIFY_EMAIL':
+    case 'ENTER_AMOUNT':
+      return {
+        cryptoCurrency: payload.cryptoCurrency,
+        fiatCurrency: payload.fiatCurrency,
+        method: payload.method,
+        orderType: payload.orderType || 'BUY',
+        pair: payload.pair,
+        step: payload.step,
+        swapAccount: payload.swapAccount
+      }
+    case 'CRYPTO_SELECTION':
+      return {
+        cryptoCurrency: payload.cryptoCurrency,
+        fiatCurrency: payload.fiatCurrency,
+        orderType: payload.orderType,
+        step: payload.step
+      }
+    case 'BANK_WIRE_DETAILS':
+      return {
+        addBank: payload.addBank,
+        displayBack: payload.displayBack,
+        fiatCurrency: payload.fiatCurrency,
+        step: payload.step
+      }
+    case 'PREVIEW_SELL': {
+      return { sellOrderType: payload.sellOrderType, step: payload.step }
+    }
+    case 'AUTHORIZE_PAYMENT':
+    case 'CHECKOUT_CONFIRM':
+    case 'ORDER_SUMMARY':
+    case 'OPEN_BANKING_CONNECT':
+      return { order: payload.order, step: payload.step }
+    case '3DS_HANDLER':
```

Comment:
since this case returns the same thing as the previous case can you just add `3DS_HANDLER` to the list above?

Reply:
Because it seems that I open a rabbit role of types that I can't fix if I do this, I didn't touch this. It was just auto-fixed.